tests/column_resizer_tester.ts

```diff
@@ -0,0 +1,264 @@
+      resizer.onMouseDown(event);
+      event = new MouseEvent(170, 50, -20) as any;
+      resizer.onMouseMove(event);
+      event = new MouseEvent(170, 50, 0) as any;
+      resizer.onMouseUp(event);
+      event = new MouseEvent(170, 60, 0) as any;
+      resizer.onMouseDown(event);
+      event = new MouseEvent(150, 60, -20) as any;
+      resizer.onMouseMove(event);
+      event = new MouseEvent(150, 60, 0) as any;
+      resizer.onMouseUp(event);
+      Expect(table.getColumnWidth(0)).toEqual(160);
+      Expect(table.getColumnWidth(1)).toEqual(300);
+      Expect(table.getColumnWidth(2)).toEqual(200);
+      Expect(table.corners.bottomRight.x).toEqual(660);
+  }
+
+  /** Tests what happens when a mousemove happens if a mousedown is called 
+   * outside the resize region and moved into resize region. 
+   */
+  @Test()
+  public mouseClickedOutsideAndMoved(): void {
+    const table = new FauxTableInterface();
+      const resizer = new ColumnResizer(table);
+      let event = new MouseEvent(800, 300, 0) as any;
+      resizer.onMouseDown(event);
+      event = new MouseEvent(190, 50, -610) as any;
+      resizer.onMouseMove(event);
+      event = new MouseEvent(190, 50, 0) as any;
+      resizer.onMouseUp(event);
+      Expect(table.getColumnWidth(0)).toEqual(200);
+      Expect(table.getColumnWidth(1)).toEqual(300);
+      Expect(table.getColumnWidth(2)).toEqual(200);
+      Expect(table.corners.bottomRight.x).toEqual(700);
+  }
+
+  /** Tests what happens when the mouse is moved without mousedown events. */
+  @Test()
+  public testMoveNoMouseDown(): void {
+    const table = new FauxTableInterface();
+    const resizer = new ColumnResizer(table);
+    let event = new MouseEvent(800, 300, 50) as any;
+    resizer.onMouseMove(event);
+    event = new MouseEvent(190, 50, -30) as any;
+    resizer.onMouseMove(event);
+    Expect(table.getColumnWidth(0)).toEqual(200);
+    Expect(table.getColumnWidth(1)).toEqual(300);
+    Expect(table.getColumnWidth(2)).toEqual(200);
+    Expect(table.corners.bottomRight.x).toEqual(700);
+  }
+
+  /** Tests what happens when two different columns are resized */
+  @Test()
+  public testTwoResizes(): void {
+    const table = new FauxTableInterface();
+    const resizer = new ColumnResizer(table);
+    let event = new MouseEvent(492, 20, 0) as any;
+    resizer.onMouseDown(event);
+    event = new MouseEvent(482, 50, -10) as any;
+    resizer.onMouseMove(event);
+    event = new MouseEvent(482, 50, 0) as any;
+    resizer.onMouseUp(event);
+    event = new MouseEvent(680, 20, 0) as any;
+    resizer.onMouseDown(event);
+    event = new MouseEvent(710, 50, 30) as any;
+    resizer.onMouseMove(event);
+    event = new MouseEvent(710, 50, 0) as any;
+    resizer.onMouseUp(event);
+    Expect(table.getColumnWidth(0)).toEqual(200);
+    Expect(table.getColumnWidth(1)).toEqual(290);
+    Expect(table.getColumnWidth(2)).toEqual(230);
+    Expect(table.corners.bottomRight.x).toEqual(720);
+  }
```

Comment:
What happens if the user resizes more than the width of the column? For example the column has a width of 100 pixels and then the user resizes the column by moving the mouse to the left more than 100 pixels.